src/rich-text/commands/index.ts

```diff
@@ -96,6 +96,51 @@ function toggleBlockType(
     };
 }
 
+const headingInfo = {
+    level: 0,
+    time: 0,
+};
+
+export function cycleThroughHeadings(nodeType: NodeType) {
+    return (state: EditorState, dispatch: (tr: Transaction) => void) => {
+        const delay = Date.now() - headingInfo.time;
```

Comment:
I'd prefer to not check for delays in our underlying code. I think this is to support successive presses causing cycling, but single presses causing toggle? If that's something we'd like to pursue, let's move that logic into the keybind file instead